oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/OAuth2AuthorizationCodeGrantFilter.java

```diff
@@ -166,7 +171,8 @@ private void processAuthorizationResponse(HttpServletRequest request, HttpServle
 
 		MultiValueMap<String, String> params = OAuth2AuthorizationResponseUtils.toMultiMap(request.getParameterMap());
 		String redirectUri = UriComponentsBuilder.fromHttpUrl(UrlUtils.buildFullRequestUrl(request))
-				.replaceQuery(null)
+				.replaceQueryParam(OAuth2ParameterNames.CODE)
```

Comment:
@ttddyy After reviewing the spec:

> The redirection endpoint URI MUST be an absolute URI as defined by
   [RFC3986] Section 4.3.  The endpoint URI MAY include an
   "application/x-www-form-urlencoded" formatted (per Appendix B) query
   component ([RFC3986] Section 3.4), which MUST be retained when **adding
   additional query parameters**.  The endpoint URI MUST NOT include a
   fragment component.

NOTE the bold highlight. As per spec, the Authorization Server will ADD additional query parameters to the `redirect-uri`. This means that the query parameters in the `redirect_uri` (if any) should be in the correct order. Also, in addition to `code` and `state`, the Authorization Server may add additional query parameters. I've tested this with google and additional parameters are added on top of `code` and `state`. Can you please add a test for this scenario and adjust the code for this.